### PR TITLE
feat(deployer, gateway): remove deployer proxy

### PR DESCRIPTION
**Pull request by jonaro00**
## Description of change

- Removes the proxy layer (including project name guard and fqdn check) from deployer.
  - Reduces deployer binary's CPU usage a lot.
- Runtimes always run on `0.0.0.0:8000` in deployer. Incoming proxy traffic hits the user app instantly.
- Address field in deployments table is always NULL from now on.
- Cleanup/refactor

Things to verify:
- Do old deployers break due to potential missing arg?
  - I assume not, since the args are only used at creation of a container

## How has this been tested? (if applicable)

Local stack successfully deploys, stops, and gets deployment summary.
Request throughput from a deployed app increased significantly.
Server header being set by gateway proxy works.

### CODE REVIEW

File: deployer/src/persistence/mod.rs [hunk 1]
```diff
@@ -210,7 +208,7 @@ impl Persistence {
             .bind(deployment.service_id.to_string())
             .bind(deployment.state)
             .bind(deployment.last_update)
-            .bind(deployment.address.map(|socket| socket.to_string()))
+            .bind(Option::<String>::None)
```
Comment (chesedo) [inline deployer/src/persistence/mod.rs:184]: Why don't we just remove the address field from the DB?

File: deployer/tests/integration_run.rs [hunk 2]
```diff
@@ -154,10 +154,6 @@ struct StubDeploymentUpdater;
 impl DeploymentUpdater for StubDeploymentUpdater {
```
Comment (chesedo) [inline deployer/tests/integration_run.rs:154]: This whole stub is now unused right?

File: gateway/src/api/latest.rs [hunk 3]
```diff
@@ -363,35 +364,67 @@ async fn delete_project(
...
 
-    // Check if it matches route: "/projects/:project_name/services/:project_name"
-    if method == Method::POST
-        && uri_path == format!("/projects/{}/services/{}", project_name, project_name)
-    {
-        let account_name = scoped_user.user.claim.sub.clone();
+        if let Err(err) = posthog_client.capture(event).await {
+            error!(error = %err, "failed to send event to posthog")
+        };
+    });
```
Comment (chesedo) [inline gateway/src/api/latest.rs:380]: Thanks for this cleanup! :pray:

**Review by chesedo — COMMENTED** [review]
Looking real good. Have a few small questions

File: deployer/src/deployment/run.rs [hunk 4]
```diff
@@ -410,24 +397,14 @@ async fn load(
...
     mut runtime_client: RuntimeClient<ClaimService<InjectPropagation<Channel>>>,
     address: SocketAddr,
-    deployment_updater: impl DeploymentUpdater,
     cleanup: impl FnOnce(Option<SubscribeStopResponse>) + Send + 'static,
 ) {
-    if let Err(err) = deployment_updater.set_address(&id, &address).await {
-        // Clean up based on a stop response built outside the runtime
-        cleanup(Some(SubscribeStopResponse {
-            reason: StopReason::Crash as i32,
-            message: format!("errored while setting the new deployer address: {}", err),
```
[the remark this answers is not in the document]
Comment (jonaro00) [reply deployer/src/deployment/run.rs:425]: Yes PLEASE

File: deployer/src/handlers/mod.rs [hunk 5]
```diff
@@ -60,15 +58,14 @@ pub struct PaginationDetails {
 #[derive(Clone)]
 pub struct RouterBuilder {
     router: Router,
-    project_name: ProjectName,
+    _project_name: ProjectName,
```
[the remark this answers is not in the document]
Comment (jonaro00) [reply deployer/src/handlers/mod.rs:61]: I feel like we should need it, so I'm keeping this one around. Fixing project<->service name sync might benefit from it.

File: deployer/src/persistence/mod.rs (hunk 1 above)
Comment (jonaro00) [reply deployer/src/persistence/mod.rs:184]: *"for historical purposes..."*
(My thinking is that they'll all get cleaned up in future migrations anyways.)

File: deployer/src/handlers/mod.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (chesedo) [reply deployer/src/handlers/mod.rs:61]: Oky doky :+1:

File: deployer/src/persistence/mod.rs (hunk 1 above)
Comment (chesedo) [reply deployer/src/persistence/mod.rs:184]: Haha, I was thinking that removing them now makes future migrations (especially the big one) easier :laughing:

File: deployer/tests/integration_run.rs [hunk 6]
```diff
@@ -147,22 +144,6 @@ impl ResourceManager for StubResourceManager {
...
-
-#[async_trait]
-impl DeploymentUpdater for StubDeploymentUpdater {
-    type Err = std::io::Error;
-
-    async fn set_address(&self, _id: &Uuid, _address: &SocketAddr) -> Result<(), Self::Err> {
-        Ok(())
-    }
-
-    async fn set_is_next(&self, _id: &Uuid, _is_next: bool) -> Result<(), Self::Err> {
```
Comment (chesedo) [inline deployer/tests/integration_run.rs:161]: What happened with this method?

**Review by chesedo — COMMENTED** [review]
Just one nit left

**Review by chesedo — APPROVED** [review]
LGTM

**Pull request merged**